src/neuroglancer/datasource/computed/frontend.ts

```diff
@@ -0,0 +1,252 @@
+  }
+
+  /**
+   * Creates a ComputedVolumeDataSourceParameters object with default values
+   * populated from the first volume spec at native resolution.
+   * @param volumeSpecs ComputedVolumeSpecs
+   * @param originUrl url for the origin data source
+   * @param dataSourceProvider
+   * @returns ComputedVolumeDataSourceParameters
+   */
+  defaultParams(
+      volumeSpecs: ComputedVolumeSpecs, originUrl: string,
+      dataSourceProvider: DataSourceProvider): ComputedVolumeDataSourceParameters {
+    const spec = volumeSpecs.sources[0][0].spec;
+    // Default DataType, VolumeType, channel count
+    const {dataType, volumeType, numChannels} = volumeSpecs;
+    // Default chunk size, used for input and output computation buffer sizes.
+    const size = spec.chunkDataSize;
+    return {
+      originUrl,
+      computationParameters: {
+        inputSpec: {size: vec3.copy(vec3.create(), size), dataType, volumeType, numChannels},
+        outputSpec: {size: vec3.copy(vec3.create(), size), dataType, volumeType, numChannels}
+      },
+      inputScaleIndex: 0,
+      inputSourceIndex: 0,
+      outputVoxelSize: vec3.copy(vec3.create(), spec.voxelSize),
+      outputVolumeSize: vec3.copy(vec3.create(), spec.upperVoxelBound),
+      outputVoxelOffset: vec3.copy(vec3.create(), spec.baseVoxelOffset),
+      dataSourceProvider
+    };
+  }
+
+  get description() {
+    return 'Computed data source.';
+  }
+
+  getVolume(
+      chunkManager: ChunkManager, config: string, options: GetVolumeOptions,
+      cancellationToken: CancellationToken) {
+    // Config is expected to be a json string, for example:
+    //   {"origin":"brainmaps://p:d:v","computation":"example","inputSize":
+    //     [36,36,32],"outputSize":[32,32,32]}
+    console.log('Computed datasource config:', config);
+    if (!options.dataSourceProvider) {
+      return Promise.reject(new Error('Need a DataSourceProvider'));
+    }
+
+    const dataSourceProvider = options.dataSourceProvider!;
+    let configObj: any;
+    try {
+      configObj = verifyObject(JSON.parse(config));
+    } catch (error) {
+      return Promise.reject(new Error(
+          `Could not parse JSON configuration while initializing computational datasource: ${
+              error}`));
+    }
+
+    if (!configObj) {
+      return Promise.reject(new Error('Could not verify configuration JSON'));
+    }
+    if (configObj['origin'] === undefined) {
+      return Promise.reject(new Error('Config is missing origin'));
+    }
+    if (configObj['computation'] === undefined) {
+      return Promise.reject(new Error('Config is missing computation'));
+    }
+
+    const computationName = verifyString(configObj['computation']);
+    const computationProvider = ComputedDataSource.computationMap.get(computationName);
+
+    const originUrl = verifyString(configObj['origin']);
+
+    if (!computationProvider) {
+      return Promise.reject(new Error(`Unable to find computation ${computationName}`));
+    }
+
```

Comment:
Add memoization based on the config, using chunkManager.memoize.getUncounted.  Otherwise you'll lose the cached data too easily, as layers can be reloaded from their JSON representation frequently (e.g. when using the Python API).  Refer to e.g. datasource/brainmaps/frontend.ts:getVolume

Reply:
Done.